src/D_functions_and_procedures.sql

```diff
@@ -362,4 +362,192 @@ $$ LANGUAGE plpgsql;
+BEGIN
+	SELECT ARRAY_AGG(DISTINCT cm.person_id) INTO actors
+	FROM cast_member cm
+	WHERE cm.media_id = input_media_id;
+	
+	CREATE TEMPORARY TABLE similar_cast (
+	media_id INTEGER,
+	"count" INTEGER
+	) ON COMMIT DROP;
+	
+	FOREACH actor IN ARRAY COALESCE(actors, '{}')
+	LOOP
+		INSERT INTO similar_cast(media_id, "count")
+		SELECT DISTINCT cm.media_id, 1
+		FROM cast_member cm
+		WHERE cm.person_id = actor
+		GROUP BY cm.media_id;
+	END LOOP;
+	
+	RETURN query
+	SELECT s.media_id, count(s.media_id)::INTEGER
+	FROM similar_cast s
+	GROUP BY s.media_id;
+
+END;
+$$ LANGUAGE plpgsql;
+
+-------------------------------------------------------
+---------find_movies_from_the_same_country()-----------
+
+CREATE OR REPLACE FUNCTION find_movies_from_the_same_country(in_country_id INTEGER)
+RETURNS TABLE (
+	media_id INTEGER,
+	"count" INTEGER
+)AS $$
+BEGIN
+	RETURN query
+	SELECT mpc.media_id, count(mpc.media_id)::INTEGER
+	FROM media_production_country mpc
+	WHERE mpc.country_id = in_country_id
+	GROUP BY mpc.media_id;
+END;
+$$ LANGUAGE plpgsql;
+
+
+
+-------------------------------------------------------
+---------find_movies_with_similar_titles()-----------
+
+CREATE OR REPLACE FUNCTION find_movie_similar_titles(input_title TEXT, min_lenght INTEGER)
+RETURNS TABLE (
+	media_id INTEGER,
+	"count" INTEGER
+)AS $$
+DECLARE
+	word TEXT;
+	title_words TEXT[];
+BEGIN
+	CREATE TEMPORARY TABLE similar_title (
+		media_id INTEGER,
+		"count" INTEGER
+		) ON COMMIT DROP;
+	
+	title_words := string_to_array(input_title, ' ');
+	
+	FOREACH word IN ARRAY COALESCE(title_words, '{}')
+	LOOP
+		IF LENGTH(word) > min_lenght THEN
+			INSERT INTO similar_title
+			SELECT DISTINCT r.media_id, 1 AS "count"
+			FROM "release" r
+			WHERE r.title ILIKE '%' ||word|| '%';
+			RAISE NOTICE 'Word: % (Length: %)', word, LENGTH(word);
+		END IF;
+	END LOOP;
+	
+	RETURN query
+	SELECT s.media_id, count(s."count")::INTEGER as count
+	FROM similar_title s
+	GROUP BY s.media_id
+	ORDER BY count DESC;
+END;
+$$ LANGUAGE plpgsql;
+
+-------------------------------------------------------
+---------find_movies_with_similar_crew()-----------
+CREATE OR REPLACE FUNCTION find_movies_with_similar_crew(input_media_id INTEGER)
+RETURNS TABLE(
+	media_id INTEGER,
+	count INTEGER
+) AS $$
+DECLARE
+	"member" INTEGER;
+	members INTEGER[];
+BEGIN
+	SELECT ARRAY_AGG(DISTINCT cm.person_id) INTO members
+	FROM crew_member cm
+	WHERE cm.media_id = input_media_id;
+	
+	CREATE TEMPORARY TABLE similar_crew (
+	media_id INTEGER,
+	"count" INTEGER
+	) ON COMMIT DROP;
+	
+	FOREACH "member" IN ARRAY COALESCE(members, '{}')
+	LOOP
+		INSERT INTO similar_crew(media_id, "count")
+		SELECT DISTINCT cm.media_id, 1
+		FROM crew_member cm
+		WHERE cm.person_id = "member"
+		GROUP BY cm.media_id;
+	END LOOP;
+	
+	RETURN query
+	SELECT s.media_id, count(s.media_id)::INTEGER
+	FROM similar_crew s
+	GROUP BY s.media_id;
+END;
+$$ LANGUAGE plpgsql;
+
+--------------------------------------------------
```

Comment:
Could be nice to have under each function, then I at least understand what is what  - but nice work.